CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs

```diff
@@ -556,7 +605,16 @@ public PromptDouble(PromptOptions<double> promptOptions)
 
             protected override bool TryParse(IMessageActivity message, out double result)
             {
-                return double.TryParse(message.Text, out result);
+                var options = new PromptRecognizeNumbersOptions { IntergerOnly = false };
+                var matches = PromptRecognizers.RecognizeNumbers(message, options);
```

Comment:
>PromptRecognizers [](start = 30, length = 17)

If RecognizeX(...) functions become member methods instead of static functions, PrompRecognizer can become part of prompt options and the default prompt options can use a PromptRecognizer configured with the default options.